javascore/nativecoinIRC2/src/main/java/foundation/icon/btp/nativecoinIRC2/NativeCoinService.java

```diff
@@ -0,0 +1,579 @@
+            //  If receiving a RES_UNKNOWN_TYPE, ignore this message
+            //  or re-send another correct message
+            UnknownResponse(_from, _sn);
+        } else {
+            //  If none of those types above, BSH responds a message of RES_UNKNOWN_TYPE
+            TransferResponse response = new TransferResponse();
+            response.setCode(TransferResponse.RC_ERR);
+            response.setMessage(TransferResponse.ERR_MSG_UNKNOWN_TYPE);
+            sendMessage(_from, NCSMessage.UNKNOWN_TYPE, _sn, response.toBytes());
+        }
+    }
+
+    @External
+    public void handleBTPError(String _src, String _svc, BigInteger _sn, long _code, String _msg) {
+        require(Context.getCaller().equals(bmc), "Only BMC");
+        TransferResponse response = new TransferResponse();
+        response.setCode(TransferResponse.RC_ERR);
+        response.setMessage("BTPError [code:" + _code + ",msg:" + _msg);
+        handleResponse(_sn, response);
+    }
+
+    @External
+    public void handleFeeGathering(String _fa, String _svc) {
+        require(Context.getCaller().equals(bmc), "Only BMC");
+        BTPAddress from = BTPAddress.valueOf(_fa);
+        Address owner = Context.getAddress();
+
+        List<String> coinNames = new ArrayList<>();
+        List<BigInteger> feeAmounts = new ArrayList<>();
+        for (String coinName : coinNames()) {
+            BigInteger feeAmount = clearFee(coinName);
+            if (feeAmount.compareTo(BigInteger.ZERO) > 0) {
+                coinNames.add(coinName);
+                feeAmounts.add(feeAmount);
+            }
+        }
+
+        if (coinNames.size() > 0) {
+            if (from.net().equals(net)) {
+                Address fa = Address.fromString(from.account());
+                int idx = coinNames.indexOf(name);
+                if (idx >= 0) {
+                    coinNames.remove(idx);
+                    BigInteger feeAmount = feeAmounts.remove(idx);
+                    Context.transfer(fa, feeAmount);
+                }
+                _transferBatch(fa, ArrayUtil.toStringArray(coinNames), ArrayUtil.toBigIntegerArray(feeAmounts));
+            } else {
+                sendRequest(owner, from, coinNames, feeAmounts);
+            }
+        }
+    }
+
+    private void _transferBatch(Address to, String[] coinNames, BigInteger[] amounts) {
+        logger.println("transferBatch", to, StringUtil.toString(coinNames), StringUtil.toString(amounts));
+        for (int i = 0; i < coinNames.length; i++) {
+            _transfer(irc2, to, amounts[i]);
+        }
+    }
+
+    private Balance getBalance(String coinName, Address owner) {
+        Balance balance = balances.at(coinName).get(owner);
+        if (balance == null) {
+            balance = new Balance(BigInteger.ZERO, BigInteger.ZERO, BigInteger.ZERO);
+            return balance;
+        }
+        Balance newBal = new Balance(balance.getUsable(), balance.getLocked(), balance.getRefundable());
```

Comment:
Is it necessary to new Balance instance here?

Reply:
While doing javascore unit testing, without the creation of new instance, it updated all previous instances of balances in the test lines while executing. I wasn't able to exactly pointout why, but this fixed the issue